tools/amplify-preview/main.go

```diff
@@ -0,0 +1,136 @@
+/*
+Copyright 2024 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"context"
+	"errors"
+	"log/slog"
+	"os"
+	"strings"
+	"time"
+
+	"github.com/alecthomas/kingpin/v2"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/aws/retry"
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/service/amplify"
+	"github.com/aws/aws-sdk-go-v2/service/amplify/types"
+)
+
+var (
+	logger = slog.New(slog.NewTextHandler(os.Stderr, &slog.HandlerOptions{Level: slog.LevelDebug}))
+
+	amplifyAppIDs  = kingpin.Flag("amplify-app-ids", "List of Amplify App IDs").Envar("AMPLIFY_APP_IDS").Required().Strings()
+	gitBranchName  = kingpin.Flag("git-branch-name", "Git branch name").Envar("GIT_BRANCH_NAME").Required().String()
+	createBranches = kingpin.Flag("create-branches",
+		"Defines whether Amplify branches should be created if missing, or just lookup existing ones").Envar("CREATE_BRANCHES").Default("false").Bool()
+	wait = kingpin.Flag("wait",
+		"Wait for pending/running job to complete").Envar("WAIT").Default("false").Bool()
+)
+
+const (
+	jobWaitSleepTime    = 30 * time.Second
+	jobWaitTimeAttempts = 40
+)
+
+func main() {
+	kingpin.Parse()
+	ctx := context.Background()
+
+	cfg, err := config.LoadDefaultConfig(context.TODO(), config.WithRetryer(func() aws.Retryer {
+		return retry.AddWithMaxAttempts(retry.NewStandard(), 10)
+	}))
+	if err != nil {
+		logger.Error("failed to load configuration", "error", err)
+		os.Exit(1)
+	}
+
+	amp := AmplifyPreview{
+		client: amplify.NewFromConfig(cfg),
+		appIDs: func() []string {
+			if len(*amplifyAppIDs) == 1 {
+				// kingpin env variables are separated by new lines, and there is no way to change the behavior
+				// https://github.com/alecthomas/kingpin/issues/249
+				return strings.Split((*amplifyAppIDs)[0], ",")
+			}
+			return *amplifyAppIDs
+		}(),
```

Comment:
Try to avoid two things: dereferencing pointers & inlined auto-executing functions. They are buggy and difficult to follow.

Cobra may handle the newlines issue better by using its `Slice` flags funcs, and pointers are not needed when using its `Var` flag functions. That may remove the need for the inlined func call, otherwise I'd just define a new variable above this AmplifyPreview init that creates the app ids slice and when assign it here.